From 7c18aedb38a3317e5fd1a855d1f7411e00fe8ebd Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Tue, 30 Nov 2021 14:05:42 +0800
Subject: [PATCH] nnapi: optional build cts of nnapi

    Add support to build cts of nnapi
    The Makefiles are based on file nn/chromeos/tests/BUILD.gn

    Signed-off-by: kidd <kidd-kw.chen@mediatek.com>

%% original patch: 0001-nnapi-enable-cts-of-nnapi.patch
---
 Makefile                                 |  42 +-
 nn/Makefile                              |  26 +
 nn/chromeos/testrunner.cpp               |   2 +-
 nn/runtime/test/GeneratedTestUtils.cpp   |   3 +
 nn/runtime/test/Makefile                 | 669 +++++++++++++++++++++++
 nn/runtime/test/fuzzing/RandomVariable.h |   1 +
 6 files changed, 718 insertions(+), 25 deletions(-)
 create mode 100644 nn/Makefile
 create mode 100644 nn/runtime/test/Makefile

diff --git a/Makefile b/Makefile
index 9b17b0d8a..1989d3462 100644
--- a/Makefile
+++ b/Makefile
@@ -1,37 +1,31 @@
 ROOT_DIR = $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
-LIBDIR ?= /usr/lib/
-INCLUDEDIR ?= /usr/include/
-
-TARGET = libneuralnetworks.so

 OUT=$(ROOT_DIR)/out
 export OUT ROOT_DIR

+BUILD_NNAPI_CTS ?= 0
+
 FLAGS := -Wall -Werror -Wextra -std=c++17 -fPIC -fvisibility=default -Wno-inconsistent-missing-override -Wno-sign-compare -Wl,--no-as-needed -Wl,--no-undefined
+export FLAGS

 DEFINES = -D __ANDROID_API__=30 -D __ANDROID_API_O_MR1__=27 -D __ANDROID_API_P__=28 -D __ANDROID_API_Q__=29 -D __ANDROID_API_R__=30 -D __ANDROID_API_FUTURE__=10000 -D NNAPI_CHROMEOS -D TF_LITE_DISABLE_X86_NEON -D NN_INCLUDE_CPU_IMPLEMENTATION
+export DEFINES

-INCLUDES = -I $(LIBTEXTCLASSIFIER_INCLUDE) -I $(EIGEN_INCLUDE) -I $(GEMMLOWP_INCLUDE) -I $(AOSP_INCLUDE) -I $(ROOT_DIR)/nn/chromeos/includes -I $(ROOT_DIR)/nn/common/ -I $(ROOT_DIR)/nn/common/include -I $(ROOT_DIR)/nn/common/random
-
-CPPFLAGS := $(DEFINES) $(FLAGS) $(INCLUDES)
-export CPPFLAGS
-
-SUBDIRS := nn/common nn/runtime

-.PHONY : all clean install $(SUBDIRS)
-
-all : $(TARGET)
-
-$(TARGET): $(SUBDIRS)
-	$(CC) -o $(OUT)/$(TARGET) -Wl,--whole-archive $(OUT)/libruntime.a -Wl,--no-whole-archive -shared -lvendor-nn-hal -L $(OUT) -lnn-common
-
-$(SUBDIRS):
-	mkdir -p $(OUT)
-	$(MAKE) -C $@ $(MAKECMDGOALS)
+.PHONY : all clean install
+all :
+	@$(MAKE) -C nn
+ifeq ($(BUILD_NNAPI_CTS), 1)
+	@$(MAKE) -C nn/runtime/test
+endif

 install:
-	mkdir -p $(DESTDIR)/$(LIBDIR)/
-	cp $(OUT)/*.so $(DESTDIR)/$(LIBDIR)/
+	@$(MAKE) -C nn install
+ifeq ($(BUILD_NNAPI_CTS), 1)
+	@$(MAKE) -C nn/runtime/test install
+endif
+
+clean:
+	@$(MAKE) -C nn clean
+	@$(MAKE) -C nn/runtime/test clean

-clean: $(SUBDIRS)
-	rm -rf $(OUT)/$(TARGET)
diff --git a/nn/Makefile b/nn/Makefile
new file mode 100644
index 000000000..605641ff9
--- /dev/null
+++ b/nn/Makefile
@@ -0,0 +1,26 @@
+TARGET = libneuralnetworks.so
+
+INCLUDES = -I $(LIBTEXTCLASSIFIER_INCLUDE) -I $(EIGEN_INCLUDE) -I $(GEMMLOWP_INCLUDE) -I $(AOSP_INCLUDE) -I $(ROOT_DIR)/nn/chromeos/includes -I $(ROOT_DIR)/nn/common/ -I $(ROOT_DIR)/nn/common/include -I $(ROOT_DIR)/nn/common/random
+
+CPPFLAGS := $(DEFINES) $(FLAGS) $(INCLUDES)
+export CPPFLAGS
+
+SUBDIRS := common runtime
+
+.PHONY : all clean install $(SUBDIRS)
+
+all : $(TARGET)
+
+$(TARGET): $(SUBDIRS)
+	$(CC) -o $(OUT)/$(TARGET) -Wl,--whole-archive $(OUT)/libruntime.a -Wl,--no-whole-archive -shared -lvendor-nn-hal -L $(OUT) -lnn-common
+
+$(SUBDIRS):
+	mkdir -p $(OUT)
+	$(MAKE) -C $@ $(MAKECMDGOALS)
+
+install:
+	mkdir -p $(DESTDIR)/$(LIBDIR)/
+	cp $(OUT)/*.so $(DESTDIR)/$(LIBDIR)/
+
+clean: $(SUBDIRS)
+	rm -rf $(OUT)/$(TARGET)
diff --git a/nn/chromeos/testrunner.cpp b/nn/chromeos/testrunner.cpp
index d60015a06..3faf62772 100644
--- a/nn/chromeos/testrunner.cpp
+++ b/nn/chromeos/testrunner.cpp
@@ -21,7 +21,7 @@ int main(int argc, char** argv) {

   testing::InitGoogleTest(&argc, argv);
   android::base::InitLogging(argv, android::base::StderrLogger);
-  testing::GTEST_FLAG(throw_on_failure) = true;
+  //testing::GTEST_FLAG(throw_on_failure) = true;
   testing::InitGoogleMock(&argc, argv);
   return RUN_ALL_TESTS();
 }
diff --git a/nn/runtime/test/GeneratedTestUtils.cpp b/nn/runtime/test/GeneratedTestUtils.cpp
index abbf79a2b..bde2a14e3 100644
--- a/nn/runtime/test/GeneratedTestUtils.cpp
+++ b/nn/runtime/test/GeneratedTestUtils.cpp
@@ -117,6 +117,9 @@ static void createModelFromSubgraph(const TestSubgraph& subgraph, bool testDynam
             } break;
             case TestOperandLifeTime::CONSTANT_REFERENCE: {
                 const uint32_t length = operand.data.size();
+
+                ASSERT_NE(nullptr, memory->getPointer());
+
                 std::memcpy(static_cast<uint8_t*>(memory->getPointer()) + *memoryOffset,
                             operand.data.get<void>(), length);
                 model->setOperandValueFromMemory(index, memory.get(), *memoryOffset, length);
diff --git a/nn/runtime/test/Makefile b/nn/runtime/test/Makefile
new file mode 100644
index 000000000..698f32ef9
--- /dev/null
+++ b/nn/runtime/test/Makefile
@@ -0,0 +1,669 @@
+TARGET = nnapi-cts
+
+FLAGS += -Wno-reorder
+
+DEFINES += -D NN_INCLUDE_CPU_IMPLEMENTATION -D NNTEST_CTS -D NNTEST_ONLY_PUBLIC_API
+
+INCLUDES = -I $(AOSP_INCLUDE) -I $(ROOT_DIR)/nn/common/include -I $(ROOT_DIR)/nn/runtime/include/ -I $(ROOT_DIR)/nn/runtime/ -I $(ROOT_DIR)/nn/runtime/test -I $(ROOT_DIR)/nn/tools/test_generator/test_harness/include/ -I $(ROOT_DIR)/nn/driver/sample
+
+CPPFLAGS := $(DEFINES) $(FLAGS) $(INCLUDES)
+
+.PHONY : all clean install
+
+CPP_SRC := $(ROOT_DIR)/nn/chromeos/testrunner.cpp \
+	$(ROOT_DIR)/nn/runtime/test/GeneratedTestUtils.cpp \
+	$(ROOT_DIR)/nn/runtime/test/RequireDebug.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestControlFlow.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestFree.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestGenerated.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestMemory.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestNeuralNetworksWrapper.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestTrivialModel.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestUnknownDimensions.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestValidateModel.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestValidateOperations.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestValidation.cpp \
+	$(ROOT_DIR)/nn/runtime/test/TestWrapper.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/OperationManager.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/RandomGraphGenerator.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/RandomGraphGeneratorUtils.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/RandomVariable.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/TestRandomGraph.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/BoundingBox.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Broadcast.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/ConcatSplit.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Convolutions.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Elementwise.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/FullyConnected.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Normalization.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Poolings.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Reduce.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Reshape.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Resize.cpp \
+	$(ROOT_DIR)/nn/runtime/test/fuzzing/operation_signatures/Selection.cpp \
+	$(ROOT_DIR)/nn/chromeos/tests/runtime/generated_test_loader_V1_0.cc \
+	$(ROOT_DIR)/nn/chromeos/tests/runtime/generated_test_loader_V1_1.cc \
+	$(ROOT_DIR)/nn/chromeos/tests/runtime/generated_test_loader_V1_2.cc \
+	$(ROOT_DIR)/nn/chromeos/tests/runtime/generated_test_loader_V1_3.cc \
+	$(ROOT_DIR)/nn/chromeos/tests/runtime/generated_test_loader_V1_3_cts.cc \
+	$(ROOT_DIR)/nn/tools/test_generator/test_harness/TestHarness.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/add_broadcast_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/add.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/add_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_float_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_float_5.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_quant8_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/avg_pool_quant8_5.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/concat_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_1_h3_w2_SAME.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_1_h3_w2_VALID.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_3_h3_w2_SAME.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_3_h3_w2_VALID.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_channels.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_channels_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_float_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_channels.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_channels_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_overflow.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_overflow_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/conv_quant8_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depth_to_space_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depth_to_space_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depth_to_space_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depth_to_space_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depth_to_space_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_large_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_large_2_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_float_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_quant8_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_quant8_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv2d_quant8_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/depthwise_conv.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/dequantize.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/embedding_lookup.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/floor.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_float_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_quant8_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_quant8_large_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/fully_connected_quant8_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/hashtable_lookup_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/hashtable_lookup_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_normalization_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_normalization.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_normalization_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_pool_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_pool_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/l2_pool_float_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/local_response_norm_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/local_response_norm_float_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/logistic_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/logistic_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/logistic_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/logistic_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lsh_projection_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lsh_projection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lsh_projection_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm2_state2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm2_state.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm3_state2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm3_state3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm3_state.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm_state2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/lstm_state.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_float_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/max_pool_quant8_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mobilenet_224_gender_basic_fixed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mobilenet_quantized.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mul_broadcast_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mul.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mul_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/mul_relu.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu1_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu1_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu1_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu1_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu6_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu6_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu6_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu6_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/relu_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/reshape.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/reshape_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/reshape_quant8_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/reshape_weights_as_inputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/resize_bilinear_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/resize_bilinear.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/rnn.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/rnn_state.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/softmax_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/softmax_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/softmax_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/softmax_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/space_to_depth_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/space_to_depth_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/space_to_depth_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/space_to_depth_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/space_to_depth_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/svdf2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/svdf_bias_present.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/svdf.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/svdf_state.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_0/tanh.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/add_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/avg_pool_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/avg_pool_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/avg_pool_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/avg_pool_float_4_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/avg_pool_float_5_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/batch_to_space.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/batch_to_space_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/batch_to_space_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/batch_to_space_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/batch_to_space_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/concat_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/concat_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/concat_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_1_h3_w2_SAME_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_1_h3_w2_VALID_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_3_h3_w2_SAME_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_3_h3_w2_VALID_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_channels_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_channels_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_large_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_large_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/conv_float_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depth_to_space_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depth_to_space_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depth_to_space_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_large_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_large_2_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_large_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_large_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv2d_float_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/depthwise_conv_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/dequantize_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/div_broadcast_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/div_broadcast_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/div.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/div_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/embedding_lookup_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/floor_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_4d_simple.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_4d_simple_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_large_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_large_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/fully_connected_float_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/hashtable_lookup_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_normalization_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_normalization_large_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_normalization_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_pool_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_pool_float_large_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/l2_pool_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/local_response_norm_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/local_response_norm_float_4_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/logistic_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/logistic_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lsh_projection_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lsh_projection_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lsh_projection_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm2_state2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm2_state_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm3_state2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm3_state3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm3_state_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm_state2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/lstm_state_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/max_pool_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/max_pool_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/max_pool_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/max_pool_float_4_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mean_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mobilenet_224_gender_basic_fixed_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mul_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/mul_relu_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/pad.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/pad_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/pad_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/pad_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu1_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu1_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu6_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu6_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/relu_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/reshape_b138150365.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/reshape_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/reshape_weights_as_inputs_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/resize_bilinear_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/resize_bilinear_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/rnn_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/rnn_state_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/softmax_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/softmax_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_batch_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_depth_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_depth_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/space_to_depth_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_b155238914.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_omitted.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/squeeze_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_b155662254.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_10.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_10_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_11.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_11_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_4_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_5.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_5_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_6.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_6_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_7.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_7_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_8_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_9.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_float_9_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_qaunt8_10.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_qaunt8_11.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_5.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_6.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_7.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_quant8_9.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/strided_slice_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/sub_broadcast_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/sub_broadcast_float_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/sub.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/sub_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/svdf2_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/svdf_bias_present_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/svdf_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/svdf_state_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/tanh_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/transpose.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/transpose_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/transpose_float_1_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/transpose_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_1/transpose_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/abs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/add_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmax_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmax_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmax_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmax_b155660285.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmin_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmin_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmin_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/argmin_b155660285.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/avg_pool_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/axis_aligned_bbox_transform.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/batch_to_space_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bbox_graph.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_aux_input.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_cifg_peephole.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_float16_batch_major_aux_input.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_float16_batch_major.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_float16_batch_major_merge_outputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_merge_outputs.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_lstm_norm_fw_output.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/bidirectional_sequence_rnn.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/box_with_nms_limit_gaussian.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/box_with_nms_limit_hard.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/box_with_nms_limit_linear.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/cast.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/channel_shuffle.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/concat_float16_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/concat_float16_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/concat_float16_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/concat_mixed_quant.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/concat_zero_sized.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/conv2d_dilation.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/conv2d_per_channel.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/conv2d_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/depth_to_space_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/depthwise_conv2d_dilation.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/depthwise_conv2d_invalid_filter_dims.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/depthwise_conv2d_per_channel.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/depthwise_conv2d_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/dequantize_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/detection_postprocess.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/div_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/embedding_lookup_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/equal.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/expand_dims.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/exp.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/floor_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/fully_connected_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/gather.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/gather_higher_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/generate_proposals.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/greater_equal.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/greater.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/grouped_conv2d.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/heatmap_max_keypoint.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/instance_normalization.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/l2_normalization_axis.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/l2_normalization_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/l2_pool_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/layer_norm_lstm.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/less_equal.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/less.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/local_response_normalization_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/log.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/logical_and.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/logical_not.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/logical_or.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/logistic_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/log_softmax.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lsh_projection_3_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lsh_projection_4_relaxed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lsh_projection_deprecated.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lsh_projection_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm2_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm2_state2_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm2_state_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm3_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm3_state2_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm3_state3_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm3_state_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm_state2_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/lstm_state_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/maximum.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/max_pool_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/mean_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/minimum.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/mul_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/neg.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/not_equal.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_all_dims.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_low_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_low_rank_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_quant8_nonzero.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_1_float.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_1_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_all_dims.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_all_dims_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_low_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pad_v2_low_rank_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/pow.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/prelu.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/quantized_lstm.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/quantize.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/random_multinomial.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/random_multinomial_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_all_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_all.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_any_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_any.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_max_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_max.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_min_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_min.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_prod_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_prod.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_sum_b155508675.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reduce_sum.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/relu1_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/relu6_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/relu_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/reshape_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/resize_bilinear_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/resize_nearest_neighbor.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/rnn_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/roi_align.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/roi_pooling.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/rsqrt.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/select_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/sin.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/slice.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/softmax_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/space_to_batch_quant8_nonzero.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/space_to_batch_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/space_to_depth_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_float_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_float_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_float_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_float_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_float_5.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_int32_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_int32_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_int32_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_int32_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_quant8_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_quant8_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_quant8_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/split_quant8_4.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/sqrt.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/squeeze_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/strided_slice_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/strided_slice_invalid.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/sub_quantized_different_scales.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/sub_v1_2_broadcast.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/sub_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/svdf_bias_present_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/svdf_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/svdf_state_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/tanh_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/tile_1.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/tile_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/tile_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/topk_v2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/transpose_conv2d.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/transpose_conv2d_large.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/transpose_float16.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/transpose_v1_2.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_1step.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_batch_major_norm_peephole_projection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_batch_major_peephole_projection_bias.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_cifg_peephole.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_f16_batch_major.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_f16_norm_peephole_projection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_layer_norm_cifg_peephole.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_lstm_norm_peephole_projection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_2/unidirectional_sequence_rnn.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/abs_int32.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/add_int32.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/add_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/argmax_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/argmin_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/avg_pool_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/axis_aligned_bbox_transform_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/batch_to_space_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/bbox_graph_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/bidirectional_sequence_lstm.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/bidirectional_sequence_lstm_state_output.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/bidirectional_sequence_rnn_1_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/bidirectional_sequence_rnn_state_output.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/box_with_nms_limit_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/cast_identity.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/channel_shuffle_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/concat_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/conv2d_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/depth_to_space_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/depthwise_conv2d_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/dequantize_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/div_int32.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/elu.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/embedding_lookup_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/embedding_lookup_v1_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/equal_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/expand_dims_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/fill.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/fully_connected_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/gather_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/generate_proposals_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/greater_equal_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/greater_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/grouped_conv2d_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/hard_swish.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/heatmap_max_keypoint_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/if_constant.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/if_no_value.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/if_same_branch_model.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/if_simple.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/l2_normalization_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/l2_normalization_zeros.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/less_equal_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/less_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/logistic_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/maximum_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/max_pool_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/mean_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/minimum_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/mul_int32.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/mul_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/not_equal_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/pad_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/pow_same_shape.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/prelu_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/qlstm_noprojection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/qlstm_projection.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/quantize_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/reduce_max_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/reduce_min_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/relu1_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/relu6_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/relu_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/reshape_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/resize_bilinear_v1_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/resize_nearest_neighbor_v1_3.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/resize_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/roi_align_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/roi_pooling_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/select_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/slice_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/softmax_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/space_to_batch_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/space_to_depth_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/split_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/squeeze_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/strided_slice_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/sub_int32.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/sub_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/tanh_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/tile_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/topk_v2_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/transpose_conv2d_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/transpose_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/unidirectional_sequence_lstm_layer_norm_cifg_peephole_state_output.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/unidirectional_sequence_rnn.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/while_fib.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/while_infinite_loop.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/while_sum_of_powers.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/while_sum_of_powers_quant8.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3/while_sum_of_powers_quant8_signed.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/cast_mismatching_shapes.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/concat_invalid_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/conv2d_v1_3_invalid_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/if_simple_unknown_dimension.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/if_simple_unknown_rank.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/while_fib_unknown_dimension.example.cpp \
+	$(ROOT_DIR)/nn/runtime/test/generated/spec_V1_3_cts_only/while_fib_unknown_rank.example.cpp \
+
+TMP_OBJECTS := $(CPP_SRC:.cpp=.o)
+OBJECTS := $(TMP_OBJECTS:.cc=.o)
+
+all : $(TARGET)
+
+$(TARGET): $(OBJECTS)
+	$(CC) $^ -o $(OUT)/$(TARGET) -g -Wl,--whole-archive -L $(LIBBASE) -lgtest -lgmock -Wl,--no-whole-archive -lstdc++ -lrt -lpthread -lm -ltensorflowlite -lssl -lcrypto -ltextclassifier -lnnapi-support -lvendor-nn-hal -L $(OUT) -lnn-common -lneuralnetworks
+
+install:
+	mkdir -p $(DESTDIR)/$(BINDIR)/
+	cp $(OUT)/$(TARGET) $(DESTDIR)/$(BINDIR)/
+
+clean:
+	rm -rf $(OBJECTS) $(OUT)/$(TARGET)
diff --git a/nn/runtime/test/fuzzing/RandomVariable.h b/nn/runtime/test/fuzzing/RandomVariable.h
index b9d39723d..e74b533ee 100644
--- a/nn/runtime/test/fuzzing/RandomVariable.h
+++ b/nn/runtime/test/fuzzing/RandomVariable.h
@@ -26,6 +26,7 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <limits.h>
 
 namespace android {
 namespace nn {
-- 
2.18.0

